api/v1alpha1/proxmoxcluster_types.go

```diff
@@ -50,25 +48,49 @@ type ProxmoxClusterSpec struct {
 	// +optional
 	SchedulerHints *SchedulerHints `json:"schedulerHints,omitempty"`
 
+	ClusterNetworkConfig `json:",inline"`
+}
+
+// ClusterNetworkConfig represents information about the cluster network configuration.
+type ClusterNetworkConfig struct {
 	// IPv4Config contains information about available IPV4 address pools and the gateway.
-	// this can be combined with ipv6Config in order to enable dual stack.
+	// this can be combined with ipv6Config to enable dual stack.
 	// either IPv4Config or IPv6Config must be provided.
 	// +optional
-	// +kubebuilder:validation:XValidation:rule="self.addresses.size() > 0",message="IPv4Config addresses must be provided"
-	IPv4Config *ipamicv1.InClusterIPPoolSpec `json:"ipv4Config,omitempty"`
+	IPv4Config *IPConfig `json:"ipv4Config,omitempty"`
 
 	// IPv6Config contains information about available IPV6 address pools and the gateway.
-	// this can be combined with ipv4Config in order to enable dual stack.
+	// this can be combined with ipv4Config to enable dual stack.
 	// either IPv4Config or IPv6Config must be provided.
 	// +optional
-	// +kubebuilder:validation:XValidation:rule="self.addresses.size() > 0",message="IPv6Config addresses must be provided"
-	IPv6Config *ipamicv1.InClusterIPPoolSpec `json:"ipv6Config,omitempty"`
+	IPv6Config *IPConfig `json:"ipv6Config,omitempty"`
 
 	// DNSServers contains information about nameservers used by machines network-config.
 	// +kubebuilder:validation:MinItems=1
 	DNSServers []string `json:"dnsServers"`
 }
 
+// IPConfig contains information about available IP config.
+type IPConfig struct {
+	// Addresses is a list of IP addresses that can be assigned. This set of
+	// addresses can be non-contiguous.
+	// +kubebuilder:validation:MinItems=1
```

Comment:
Shouldn't DHCP and addresses be mutually exclusive?

Reply:
DHCP and static IP addresses on teh same interface aren't mutually exclusive, and they also aren't in netplan.

Reply:
Ok, let me rephrase. Should addresses really be required when using DHCP is an option?

Reply:
No, it's mutually exclusive.

Reply:
Ok. It can't be mutually exclusive if the addresses are always expected to have one item

Reply:
That's already considered via webhook.

Reply:
Coming back to this, last week I debugged a setup which required static ip + dhcp. It's a valid usecase I guess.

Reply:
I just noticed that my initial question came from a wrong assumption that the marker would force me to always provide addresses. The behavior is already how I would've wanted it to be. Sorry for the hold up